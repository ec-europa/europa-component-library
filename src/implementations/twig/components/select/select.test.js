import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import dataSingle from '@ecl/specs-component-select/demo/data-single';
import dataMultiple from '@ecl/specs-component-select/demo/data-multiple';

expect.extend(toHaveNoViolations);

describe('Select', () => {
  const template = '@ecl/select/select.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(
        render(
          merge(dataSingle, {
            required: false,
          })
        )
      ).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(dataSingle, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraAttributes = merge(dataSingle, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(
        render(optionsWithExtraAttributes)
      ).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, dataSingle, true))
      ).toHaveNoViolations();
    });
  });

  describe('Required', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataSingle)).resolves.toMatchSnapshot();
    });
  });

  describe('Disabled', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(
        render(
          merge(dataSingle, {
            disabled: true,
            required: false,
          })
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('With error', () => {
    const options = merge(dataSingle, {
      invalid: true,
      required: false,
    });

    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(options)).resolves.toMatchSnapshot();
    });
  });

  describe('Multiple', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataMultiple)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, dataMultiple, true))
      ).toHaveNoViolations();
    });
  });
});
