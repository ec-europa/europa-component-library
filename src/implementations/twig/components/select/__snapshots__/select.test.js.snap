// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Default renders correctly 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__optional"
      >
          (optional)
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--m"
    >
      <select
        class="ecl-select"
        id="select-default"
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--m"
    >
      <select
        class="ecl-select"
        data-test="data-test-value"
        data-test-1="data-test-value-1"
        id="select-default"
        required=""
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select Default renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--m"
    >
      <select
        class="ecl-select custom-class custom-class--test"
        id="select-default"
        required=""
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select Disabled renders correctly 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label ecl-form-label--disabled"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__optional"
      >
          (optional)
      </span>
    </label>
    <div
      class="ecl-help-block ecl-help-block--disabled"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--disabled ecl-select__container--m"
    >
      <select
        class="ecl-select"
        disabled=""
        id="select-default"
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select Multiple renders correctly 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label"
      for="select-multiple"
    >
      Select a country
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--m"
    >
      <select
        class="ecl-select"
        data-ecl-auto-init="Select"
        data-ecl-select-all="Select all"
        data-ecl-select-default="Select an item"
        data-ecl-select-multiple=""
        data-ecl-select-no-results="No results found"
        data-ecl-select-search="Search"
        id="select-multiple"
        multiple=""
        required=""
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select Required renders correctly 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__required"
      >
        *
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-select__container ecl-select__container--m"
    >
      <select
        class="ecl-select"
        id="select-default"
        required=""
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Select With error renders correctly 1`] = `
<jest>
  <div
    class="ecl-form-group"
  >
    <label
      class="ecl-form-label ecl-form-label--invalid"
      for="select-default"
    >
      Select a country
      <span
        class="ecl-form-label__optional"
      >
          (optional)
      </span>
    </label>
    <div
      class="ecl-help-block"
    >
      This is the input's helper text.
    </div>
    <div
      class="ecl-feedback-message"
    >
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--m ecl-feedback-message__icon"
        focusable="false"
      >
        <use
          xlink:href="/icons.svg#error"
        />
      </svg>
            This is the error message
    </div>
    <div
      class="ecl-select__container ecl-select__container--invalid ecl-select__container--m"
    >
      <select
        class="ecl-select"
        id="select-default"
      >
        <option
          value="1"
        >
          Belgium
        </option>
        <option
          value="2"
        >
          France
        </option>
        <option
          disabled=""
          value="3"
        >
          Luxembourg
        </option>
        <option
          value="4"
        >
          Germany
        </option>
        <option
          selected=""
          value="5"
        >
          Bulgaria
        </option>
        <option
          value="6"
        >
          Italy
        </option>
        <option
          value="7"
        >
          Romania
        </option>
        <option
          value="8"
        >
          Greece
        </option>
        <option
          value="9"
        >
          Hungary
        </option>
        <option
          value="10"
        >
          Portugal
        </option>
      </select>
      <div
        class="ecl-select__icon"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </div>
    </div>
  </div>
</jest>
`;
